src/Particle/SampleStack.cpp

```diff
@@ -88,30 +38,37 @@ void SampleStack::setMaxSamples(int n)
   }
 }
 
-/** save the [first,last) walkers to SampleStack
- */
-void SampleStack::saveEnsemble(walker_iterator first, walker_iterator last)
+MCSample& SampleStack::getSample(unsigned int i) const { return *sample_vector_[i]; }
+
+void SampleStack::saveEnsemble(std::vector<MCSample>& walker_list)
 {
   //safety check
   if (max_samples_ == 0)
     return;
+  auto first = walker_list.begin();
+  auto last  = walker_list.end();
   while ((first != last) && (current_sample_count_ < max_samples_))
   {
-    sample_vector_[current_sample_count_]->put(**first);
+    *sample_vector_[current_sample_count_] = *first;
     ++first;
     ++current_sample_count_;
   }
 }
 
+void SampleStack::appendSample(MCSample &&sample)
+{
+  if (current_sample_count_ < max_samples_)
+  {
+    *sample_vector_[current_sample_count_] = std::move(sample);
+    current_sample_count_++;
+  }
```

Comment:
Prudent to add an error abort here if current_sample_count_ >= max_samples_?

Reply:
We allow nsamples < number of samples generated by VMC. The overflow portion is thrown away.

Reply:
Then a comment to indicate that this is where this application logic/silent feature exists and is implemented.

Reply:
The place to check this and warn the user would be in VMCBatched::calc_default_local_walkers.